### ISSUE TITLE
BBN LOGO

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#1230]
This source file:  
https://github.com/PDP-10/bbn-logo

Assembles cleanly with MACRO, and loads with LOADER.  It then runs fine under DECUUO:

```
LOGO

_PRINT 42
42
_

**Comment by larsbrinkhoff** [comment]
@Alexey-Slyusar, [this is now built on the lars/bbnlogo branch](https://github.com/PDP-10/its/blob/240a5ca3806606ed32b8995194fa20566f7e6067/build/misc.tcl#L1157-L1168).

Run it from ITS with the command `:BBN;LOGO`.

**Comment by larsbrinkhoff** [comment]
@Alexey-Slyusar, you are right is behaves very much like CLOGO.  Even the GB command replies "AND A PLEASANT DAY TO YOU!"

**Comment by larsbrinkhoff** [comment]
I take back my previous statement that this Logo isn't similar internally to CLOGO.  I now found several symbols and code sections that looks to be the same.  So we can now confidently say CLOGO is a descendant of this "BBN" Logo.

**Comment by Alexey-Slyusar** [comment]
Great! Thank you!!!
So now everyone has a possibility to touch the very origins of the Logo
culture! Hooray!!!
I can't wait to try things out!
Unfortunately I can't do it immediately and have to languish :-)

I assume that you built BBN LOGO from source code and it is not just a
binary. Could you please point where the SAVE and GET procedures definition
may be found.

Thanks again!!!



вт, 18 сент. 2018 г. в 11:18, Lars Brinkhoff <notifications@github.com>:

**Comment by larsbrinkhoff** [comment]
The very first origin would be PDP-1 Logo, wouldn't it?

Yes, it's from MACRO-10 source code.

SAVE and GET in this TOPS-10 version maybe aren't so illuminating for understanding CLOGO.  But here they are:
- [SAVE](https://github.com/PDP-10/its/blob/240a5ca3806606ed32b8995194fa20566f7e6067/src/bbn/logo.mac#L5578-L5753)
- [GET](https://github.com/PDP-10/its/blob/240a5ca3806606ed32b8995194fa20566f7e6067/src/bbn/logo.mac#L5470-L5576)

**Comment by Alexey-Slyusar** [comment]
Oh I am glad that the hypothesis about BBN LOGO and CLOGO similarity was
confirmed. So despite the fact that we don't have the CLOGO source code we
can use BBN LOGO code as reference. Along with the SAVE and GET procedures
I very interested in the BURY and DIGUP ones.

вт, 18 сент. 2018 г. в 11:54, Алексей Слюсарь <aleksey.slyusar@gmail.com>:

**Comment by Alexey-Slyusar** [comment]
Oh, for sure, Lars, you are defenitely right. the very first origin would
be PDP-1 Logo. I was not accurate.

At the same time, the BBN LOGO is the first Logo version with relatively
comprehensive documentation. And everyone who interested may take the
documentation and fill the spirit of the  language and pedagodgical ideas
circa 1969 that very close to the origins.


вт, 18 сент. 2018 г. в 12:13, Алексей Слюсарь <aleksey.slyusar@gmail.com>:

**Comment by Alexey-Slyusar** [comment]
It is strange that BURY and DIGUP don't appear in the source file.
Here is the sreenshot from the BBN Report No. 1889:
![image](https://user-images.githubusercontent.com/42380964/45702476-9aa97080-bb7a-11e8-9a2f-dda611ed977c.png)
By the way, do you have any ideas about possible meaning for (WHEEL ONLY)? Is it system related (I mean ITS or TENEX or any other OS for PDP machines) or just Logo related word?

**Comment by larsbrinkhoff** [comment]
"Wheel" usually means "privileged group" in an operating system.  I don't know if this applies to TENEX which is the BBN operating system.

**Comment by Alexey-Slyusar** [comment]
I haven't  got the BBN LOGO running:

```
:CWD BBN;

BBN;LOGO
DSK: BBN; TS LOGO - FILE NOT FOUND
:LISTF

DB   BBN
FREE BLOCKS #0=18684
  0   LOGO   MAC    37   9/18/2018 12:28:38
  0   LOGO   REL    11   9/18/2018 20:48:32
  0   TS     ADVENT 10   9/18/2018 20:48:35
```
What did I do incorrect?

**Comment by eswenson1** [comment]
I think you forgot to create an executable image. You have to link/load the .REL file and create an TS LOGO file.  I think these are the steps:

```
*:dec sys:loader
*logo/go

LOGO 1+7K CORE, 783+633 WORDS FREE
LOADER USED 9+4K CORE
[NULL LOW SEGMENT]
EXIT
*:start 45

Command: dump
*:pdump ts logo
*:kill
```

**Comment by larsbrinkhoff** [comment]
It's my fault!  In my commit, the :pdump command writes to ts advent, not ts logo as intended.  Copy-paste error.

**Comment by larsbrinkhoff** [comment]
@Alexey-Slyusar, you can just follow Eric's intructions to make the program.  You don't have to rebuild from scratch.

**Comment by eswenson1** [comment]
Or he could just run BBN; ADVENT, right?  :-)

**Comment by larsbrinkhoff** [comment]
No!  That's just plain wrong.

**Comment by larsbrinkhoff** [comment]
No use risking getting eaten by a grue.

**Comment by eswenson1** [comment]
You'd better be careful, I might flip the magic switch.

**Comment by Alexey-Slyusar** [comment]
Oh, no! It seems that you did that!!!

`DRAGON;DRAGON HOARD  Pack 0., Unit #0 Shares block with some other file
Tracking down shared blocks.
.MAIL.	;  LIST  QUEUE  0-36588, TUT=0
DRAGON	;DRAGON HOARD  0-36588, TUT=0
*** ERROR *** THE SYSTEM MAY NOT BE BROUGHT UP
`

**Comment by Alexey-Slyusar** [comment]
:-)

**Comment by Alexey-Slyusar** [comment]
Flip the magic switch back, please!

**Comment by eswenson1** [comment]
Are you serious?  The :-) called into question whether the crash was real or not...

**Comment by Alexey-Slyusar** [comment]
Hi Eric!

Yes, my system crashed. And it is sad of course.
But the good news is that the magic switch seems to somehow work.

**Comment by larsbrinkhoff** [comment]
Maybe it's my fault for adding the switch back to the KA10 emulator.  It has been lying dormant for decades, now it's back in full force, more potent than ever!

**Comment by Alexey-Slyusar** [comment]
Oh, I realised that the situation is very serious. It is not just a switch and there are elven inscriptions on it. So this is not just a joke. 
Is It necessary to melt the switch in the Orodruin's Cracks of Doom?
Or will we be able to use that force for the good?

**Comment by Alexey-Slyusar** [comment]
Lars, thank you for the information about WHEEL!
So in our case it is a TENEX thing. ITS has more freedom in design - there
are no privileged users, if I am not mistaken - and that is why WHEEL is
not so relevant for CLOGO.
And thanks for the the testing BBN LOGO SAVE command. So Eric's guess that
the directory is hard coded is absolutely right for BBN LOGO and most
likely right for CLOGO too. I'll keep trying to find the appropriate
directory name for storing files in CLOGO.

ср, 19 сент. 2018 г. в 3:20, Lars Brinkhoff <notifications@github.com>:

**Comment by larsbrinkhoff** [comment]
@krewat preserved this LOGO.MAC file from some ARPANET computer in 1984.  Maybe Stevens University.

So, given the comment from the top of the file, the provenance might be something like: BBN → ... → NRC → ... → Stevens → Krewat.  I'll see if I can contact A. G. Smith or R. A. Orchard.

```
;MODIFIED AT NATIONAL RESEARCH COUNCIL (ISS/REED), OTTAWA, CANADA BY A.G.SMITH
;AND R.A. ORCHARD
```

**Comment by larsbrinkhoff** [comment]
Hello @rorchard,

We are a small group researching and restoring old PDP-10 software.  We have found a copy of BBN Logo with modifications from NRC.  A comment mentions R.A. Orchard.  Are you him?  If so, do you remember using Logo?  We are trying to trace the provenance of the LOGO.MAC file we found, so any detail you might remember would be welcome.

Thank you!

**Comment by krewat** [comment]
I am the person who saved this particular LOGO.MAC ... Sadly, I do not 
have any documentation. I have only the single LOGO.MAC file.

If I have the timing right, and it might have actually been earlier than 
this that I found the file in question, I was 17 years old in 1983, 
plying the ARPANET with a 300 baud modem, and was able to audio-record 
the modem tones as I TYPE'd the file, then reversed this by playing the 
tape back to the modem set to answer (instead of originate) and transfer 
it to a Commodore 64. After that, I was able to upload it to a KS10 I 
worked on at BOCES/LIRICS in Dix Hills, NY. A year or two later, I took 
home a backup tape with my personal files on it, and kept them for 16 
years. Sometime in 2001, I was able to image those backup tapes using a 
Sun 3/280, and read those images using TOPS-10 BACKUP on simh. There is 
nothing else LOGO related in the restore.

Also, I am drawing a complete blank on where I found it. I mentioned 
somewhere in alt.sys.pdp10 circa 2001 that it might have been Stevens 
Institute of Technology, but at this point I can not count on that. 
Around 1982-83 on the ARPANET, anonymous FTP sites were everywhere, and 
I spent many many hours looking through them for goodies. It could have 
been any of a number of TOPS-10 or TOPS-20 systems present at the time.

thanks!
art k.

**Comment by larsbrinkhoff** [comment]
Hello @rorchard,

Thank you for your recollections!

**Comment by larsbrinkhoff** [comment]
Thank you @krewat, that's an interesting story in itself.  I'm impressed by your audio recording hack!

**Comment by rorchard** [comment]
I’ve looked through all my old documents and books but didn’t find anything related to the PDP-10 or Logo. 

Good luck with your work!

Bob.

**Comment by larsbrinkhoff** [comment]
Thank you @rorchard, that's very kind!

**Comment by larsbrinkhoff** [comment]
Fixed by #1233.
